### PR TITLE
Controller tests

### PR DESCRIPTION
**Pull request by IoanCosarca**
Merge controller tests into the main branch

### CODE REVIEW

File: src/test/java/com/ntt/bistroapplication/controller/v1/CustomerControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.Set;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+class CustomerControllerTest {
```
Comment (andrutza) [inline src/test/java/com/ntt/bistroapplication/controller/v1/CustomerControllerTest.java:25]: For the controller-layer classes I recommend using @WebMvcTest.

File: src/test/java/com/ntt/bistroapplication/controller/v1/CustomerControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        mockMvc = MockMvcBuilders.standaloneSetup(customerController)
+                .setControllerAdvice(new RestResponseEntityExceptionHandler())
+                .build();
+    }
+
+    @Test
+    @DisplayName(value = "Test if all the customers can be returned.")
+    void getCustomers() throws Exception
+    {
+        Set<CustomerDTO> customers = new HashSet<>();
```
Comment (andrutza) [inline src/test/java/com/ntt/bistroapplication/controller/v1/CustomerControllerTest.java]: You can extract the setup part to separate method.

File: src/test/java/com/ntt/bistroapplication/controller/v1/OrderControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        List<PlacedOrderDTO> placedOrderDTOS = new ArrayList<>();
+        PlacedOrderDTO order = new PlacedOrderDTO();
+        CustomerDTO customer = new CustomerDTO(CUSTOMER_NAME);
+        order.setCustomer(customer);
+        orderService.addOrder(order);
+        placedOrderDTOS.add(order);
+
+        when(orderService.getCustomerOrders(ID)).thenReturn(placedOrderDTOS);
+        mockMvc.perform(get(OrderController.BASE_URL + "/customerOrders/" + ID)
```
Comment (andrutza) [inline src/test/java/com/ntt/bistroapplication/controller/v1/OrderControllerTest.java]: You can extract the path to a constant.

File: src/test/java/com/ntt/bistroapplication/controller/v1/ProductControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.*;
```
Comment (andrutza) [inline src/test/java/com/ntt/bistroapplication/controller/v1/ProductControllerTest.java:20]: please try not to import everything from a certain reference, import only the resources that you need.

**Review by andrutza — CHANGES_REQUESTED** [review]

File: src/test/java/com/ntt/bistroapplication/controller/v1/ProductControllerTest.java (hunk 4 above)
Comment (IoanCosarca) [reply src/test/java/com/ntt/bistroapplication/controller/v1/ProductControllerTest.java:20]: Intellij auto-imported the resources from org.hamcrest.Matchers as I was writing the code and it considered that enough things were imported to put org.hamcrest.Matchers.*

**Review by andrutza — APPROVED** [review]

**Pull request merged**